### PR TITLE
Allow selecting the Hardware type from main.tf

### PR DESCRIPTION
**Pull request by Bischoff**
## What does this PR change?

For Retail tests in Build Validation, we want to onboard different OSes on the same branch server. The selection of the operating system to install is based on the hardware type.

Note: the new `manufacturer` and `product` provider settings are only applied to PXE boot minions. We could refactor someday to apply them to all types of hosts (and especially virtualization hosts, where they are currently hardcoded to `SUSE` and `SUSE`).

### CODE REVIEW

File: backend_modules/libvirt/host/main.tf [hunk 1]
```diff
@@ -19,8 +21,8 @@ locals {
     contains(var.roles, "virthost") ? { memory = 3072, vcpu = 3 } : {},
     contains(var.roles, "jenkins") ? { memory = 16384, vcpu = 4 } : {},
     var.provider_settings,
-    contains(var.roles, "virthost") ? { cpu_model = "host-passthrough", xslt = file("${path.module}/sysinfos.xsl") } : {},
-    contains(var.roles, "pxe_boot") ? { xslt = file("${path.module}/pxe.xsl") } : {})
+    contains(var.roles, "virthost") ? { cpu_model = "host-passthrough", xslt = file("${path.module}/virthost.xsl") } : {},
+    contains(var.roles, "pxe_boot") ? { xslt = templatefile("${path.module}/pxe_boot.xsl", { manufacturer = "${manufacturer}", product = "${product}" }) } : {})
```
Comment (srbarrios) [inline backend_modules/libvirt/host/main.tf:25]: Should that be var.manufacturer and var.product?

Comment (srbarrios) [reply backend_modules/libvirt/host/main.tf:25]: Or even maybe we should be using `var.provider_settings[manufacturer]` ?

I'm just now sure from which scope we should pick the values

File: backend_modules/libvirt/host/pxe_boot.xsl [hunk 2]
```diff
@@ -14,12 +14,12 @@
         <xsl:text>&#x0A;      </xsl:text>
         <xsl:element name="entry">
           <xsl:attribute name="name">manufacturer</xsl:attribute>
-          <xsl:text>Intel</xsl:text>
+          <xsl:text>${manufacturer}</xsl:text>
```
Comment (srbarrios) [inline backend_modules/libvirt/host/pxe_boot.xsl:17]: I'm a bit lost here, this is Jinja? We process these xsl files too?

Comment (Bischoff) [reply backend_modules/libvirt/host/pxe_boot.xsl:17]: this is HCL's `templatefile()` processing

File: backend_modules/libvirt/host/main.tf (hunk 1 above)
Comment (Bischoff) [reply backend_modules/libvirt/host/main.tf:25]: maybe yes, I'm currently on that.

Comment (Bischoff) [reply backend_modules/libvirt/host/main.tf]: Found the solution, please have a look again.

**Review by nodeg — APPROVED** [review]
LGTM.

**Review by ktsamis — APPROVED** [review]

**Review by rjmateus — COMMENTED** [review]
LGTM

**Pull request merged**